src/components/badge/index.tsx

```diff
@@ -230,38 +236,35 @@ class Badge extends PureComponent<BadgeProps> {
     }
   }
 
+  renderCustomElement() {
+    const {customElement} = this.props;
+    return customElement && <View>{customElement}</View>;
+  }
+
   renderIcon() {
-    const {icon, iconStyle, iconProps, borderColor, label} = this.props;
+    const {icon, iconStyle, iconProps, borderColor, label, customElement} = this.props;
     const flex = label ? 0 : 1;
     return (
-      <Image
-        source={icon!}
-        resizeMode="contain"
-        //@ts-ignore
-        borderColor={borderColor}
-        {...iconProps}
-        style={{
-          flex,
-          ...iconStyle
-        }}
-      />
+      !customElement &&
+      icon && (
```

Comment:
I don't think the icon should depend on customElement.. 
The user should decide what they want to pass.